### ISSUE TITLE
Named arguments for model.start_operation

### ISSUE THREAD

**Issue by thomthom** [SketchUp/api-issue-tracker#623]
Spin-off from #622

Instead of the current sequence of boolean arguments,

```rb
model.start_operation("Hello", true, false, true)
```

it would be nice to support named arguments:

```rb
model.start_operation("Hello", disable_ui: true, transparent: true)
```

This makes it easier to omit the other optional arguments. 

Note that "next_transparent" is deliberatly omitted from the example above as it's a deprecated argument we want to steer people away from. It's only there as a last resort to workaround a couple of other operation bugs.

**Comment by DanRathbun** [comment]
**I agree**.

It is also can be called on older versions (back to SU2014) as the named arguments are ignored.
 (Ie, the method just takes on the default argument settings and will not cause an `ArgumentError` exception.)

----

I would also suggest changing the result doc. I think it would always return `true` and otherwise raise either `ArgumentError` or `TypeError`. (Ie, I do not think `false` would ever be a result for this method.)

[_In hindsight, perhaps it should have just returned the model object to allow call chaining._]

**Comment by thomthom** [comment]
We could also throw in block mode while we're at it:

```rb
model.start_operation("Hello", disable_ui: true, transparent: true) do
  # Do stuff here.
end
```

Though, how to deal with aborting or commiting an operation upon errors... 🤔

```rb
model.start_operation("Hello", disable_ui: true, abort_on_error: true) do
  # Do stuff here.
end
```

Not sure if `abort_on_error` is best defaulting to `true` or `false`.

or... I might be overthinking this, it might still be doable to allow this:

```rb
model.start_operation("Hello", disable_ui: true, transparent: true) do
  # do stuff
rescue
  model.abort_operation
end
```

**Comment by DanRathbun** [comment]
> We could also throw in block mode while we're at it:

Yes I have long been an advocate of the block form for operations where the commit or abort is automatically (and appropriately) called.

This means _usually_ that the **`begin`** ... **`rescue`** is internal to the `start_operation` method and wraps the **`yield`** call. It is the whole point of making them easier for both newbs and the experienced.

I'm not sure if there is a way to detect within the method body if the passed block has an associated **`rescue`** clause ?
(Ie ... `if rescue_given?` or similar.)

----

Block form operation discussion is mostly in issue https://github.com/SketchUp/api-issue-tracker/issues/52 and a undoable block operation in issue https://github.com/SketchUp/api-issue-tracker/issues/4.
